server/chat-plugins/abuse-monitor.ts

```diff
@@ -98,6 +98,12 @@ interface BattleInfo {
 	log: string[];
 }
 
+// stolen from chatlog. necessary here, but importing chatlog sucks.
```

Comment:
Don't we already have date-handling functions in Chat? Should this also go there?

Reply:
Probably, but that should be a follow-up PR since it requires some work in chatlog.